### ISSUE TITLE
Naviagtion delay in _startOpening not working in magento 2.3.5-p1

### ISSUE THREAD

**Issue by stkrelax** [magento/magento2#28754]
### Preconditions (*)
1. magento 2.3.5-p1
2. changed topmenu.phtml --> "delay":300 added
```
<nav class="navigation bla" data-action="navigation">
    <ul data-mage-init='{"menu":{"responsive":true, "delay":300, "expanded":true, "position":{"my":"left top","at":"left bottom"}}}'>
        <?= /* @noEscape */ $_menuHtml?>
        <?= $block->getChildHtml() ?>
    </ul>
</nav>
```
 
### Steps to reproduce (*)
1. open menu via mouse over
2. close menu via mouse leave

### Expected result (*)
1. _startOpening method uses delay
2. _close uses delay

### Actual result (*)
1. _startOpening does not use delay anymore 
2. _close uses delay as provided

### Reason
this commit
https://github.com/magento/magento2/commit/747846ed0523dfa37c15655b244ed5dbe4902f92

from this "fixed" issue
https://github.com/magento/magento2/issues/25589

---
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by nkajic** [comment]
Hi @stkrelax thank you so much for reporting this reference - it saved me additional time debugging root cause to this problem.

We also had same problem and it was making menu user un-friendly since since it was missing this delay. 

Here is reference how pure jquery menu should act with delay on showing submenu. https://jqueryui.com/menu/ Two different worlds. 

Removing delay entirely was definitely not the right solution to fix it #25589

Reproduced on 2.4-develop

**Comment by stkrelax** [comment]
@nkajic you're welcome ;)
